Agrega función fillout

devarops opened the pull request:
Esta función la usamos en [`muestreo-aves-marinas-ipbc`](https://bitbucket.org/IslasGECI/muestreo-aves-marinas-ipbc/)
- 

# Requisitos

## [Implementación en Python](https://github.com/IslasGECI/manual#implementaci%C3%B3n-en-python)
- [x] Evoca mutantes de Python con Mutmut en GitHub Actions
- [x] Corre una prueba dummy con Pytest en GitHub Actions
- [x] Evalúa cambio de cobertura con Codecov en GitHub Actions
- [x] Verifica formato de Python con Black en GitHub Actions
- [x] Agrega phony lint a Makefile para correr Flake8 y Pylint (localmente, no en GitHub Actions)

## [Pruebas de Python](https://github.com/IslasGECI/manual#pruebas-de-python)
- [x] Se evocan los mutantes en GitHub Actions
- [x] No hay mutantes sobrevivientes

# Revisiones

## [1ra revisión](https://github.com/IslasGECI/distdens/pull/6#pullrequestreview-420312190) @nepito

- [x] Hay 40 mutantes sobrevivientes. Por favor atrápenlos...

## [2da revisión](https://github.com/IslasGECI/distdens/pull/6#pullrequestreview-440186347) @nepito

- [x] [Por favor refactoriza en una función las líneas 13-19, pues es código repetido](https://github.com/IslasGECI/distdens/pull/6#discussion_r447815786)
- [x] [Por favor refactoriza en una función las líneas 20-26, pues es código repetido](https://github.com/IslasGECI/distdens/pull/6#discussion_r447816130)
- [x] [Qué loco que esté teminando con una coma, ¿no?](https://github.com/IslasGECI/distdens/pull/6#discussion_r447817939)
- [x] [¿Para qué usamos esto y cómo lo obtenemos?](https://github.com/IslasGECI/distdens/pull/6#discussion_r447821428)

## [3ra revisión](https://github.com/IslasGECI/distdens/pull/6#pullrequestreview-444032951) @nepito

- [x] [muchas gracias @evaristor ya se lo agregué al repo con el que estoy trabajando](https://github.com/IslasGECI/distdens/pull/6#discussion_r450973574)
- [x] [Por favor refactoriza las líneas 8-9 y 15-24 para que la 15 quedara algo así](https://github.com/IslasGECI/distdens/pull/6#discussion_r451028652)
- [x] [Por favor refactoriza las líneas 10-11 y 25-34 para que la 25 quedara algo así](https://github.com/IslasGECI/distdens/pull/6#discussion_r451029649)

devarops committed:
Agrega función fillout

nepito reviewed (changes requested):
Hay 40 mutantes sobrevivientes. Por favor atrápenlos...

devarops committed:
Agrega primer prueba para fillout

devarops committed:
Ignora archivos pycache

devarops commented:
Quedan 9 mutantes
![image](https://user-images.githubusercontent.com/3461018/84188096-e59ea200-aa47-11ea-8dd7-3577be1419c3.png)

devarops committed:
Caza 5 mutantes

devarops commented:
Quedan 4 mutantes.

![image](https://user-images.githubusercontent.com/3461018/85171472-f0b1b900-b223-11ea-82e1-c7010ca09356.png)

devarops committed:
Formatea con Black

devarops committed:
Caza mutantes restantes

devarops committed:
Renombra objetivo mutation a mutanta

devarops commented:
> Hay 40 mutantes sobrevivientes. Por favor atrápenlos...

Listo, @nepito 

![image](https://user-images.githubusercontent.com/3461018/85810260-55cb4a00-b70f-11ea-9df6-2bab34c7fed7.png)

devarops committed:
Agrega Black y flake8

devarops committed:
Agrega formateo, linters y cobertura

devarops committed:
Verifica formateo

devarops committed:
Formatea pruebas

nepito commented on distdens/fillout.py:
```diff
@@ -0,0 +1,48 @@
+    y_inicial = np.array(limits[2]).reshape(1, 1)
+    y_final = np.array(limits[3]).reshape(1, 1)
+    indice_pegado_costa = np.where(x == min(x))[0][0]
+    x_cerrada = np.concatenate(
```
Por favor refactoriza en una función las líneas 13-19, pues es código repetido:
```suggestion
    x_cerrada = close_coordinate(x, indice_pegado_costa)
```

devarops replied:
Muchas gracias por hacer la prueba para refactorizar, @nepito. Eso es más de la mitad del trabajo.

nepito commented on distdens/fillout.py:
```diff
@@ -0,0 +1,48 @@
+            x[indice_pegado_costa].reshape(1, 1),
+        ]
+    )
+    y_cerrada = np.concatenate(
```
Por favor refactoriza en una función las líneas 20-26, pues es código repetido:
```suggestion
    y_cerrada = close_coordinate(y, indice_pegado_costa)
```

devarops replied:
Muchas gracias por hacer la prueba para refactorizar, @nepito. Eso es más de la mitad del trabajo.

nepito commented on tests/test_QuadratTableReader.py:
```diff
@@ -4,40 +4,53 @@
+    "Madrigueras_con_actividad_aparente": [0, 1, 2],
+    "Este": [374448, 374458, 374542],
+    "Norte": [3198223, 3198229, 3198266],
+    "Zona_utm": ["11R", "11R", "11R"],
```
Qué loco que esté teminando con una coma, ¿no?

devarops replied:
Quité la coma del final

devarops replied:
Pero el Black se la regresó

Commit:
Agrega prueba para refactorizar

nepito commented on Makefile:
```diff
@@ -1,12 +1,32 @@
-mutation:
-	mutmut run --paths-to-mutate distdens 
+repo = distdens
+codecov_token = eae768b1-8c32-40a8-89fd-6b7589f9efa8
```
¿Para qué usamos esto y cómo lo obtenemos?

devarops replied:
Nos muestra [esta información](https://codecov.io/gh/IslasGECI/distdens/compare/88817052a542964016fe33e817c0d94af71bb185...c18f03913267f0dad461400def51fbf7529409a9) adicional que por lo pronto podemos ignorar.
El _token_ lo obtenemos [aquí](https://codecov.io/gh/IslasGECI/distdens/settings).

nepito replied:
muchas gracias @evaristor ya se lo agregué al repo con el que estoy trabajando

nepito reviewed (changes requested):
Aquí está mi comentarios @evaristor

devarops commented:
Gracias por tu revisión @nepito

devarops committed:
Ignora carpeta distdens.egg-info/

devarops committed:
Ignora reporte de cobertura

devarops committed:
Quita coma innecesaria y ordena alfabéticamente

devarops committed:
Refactoriza creando función __close_coordinate

devarops committed:
Usa un único piso como prefijo en ligar de doble

Por que el doble se usa para métodos y tiene un significado especial mientras que el sencillo no es interpretado por Python y es para sugerir a los programadores que es para uso interno

devarops committed:
Formatea usando Black

Commit:
Cambia las pruebas a pytest

Commit:
Agrega preuba para la función privada _testing

nepito commented on distdens/fillout.py:
```diff
@@ -0,0 +1,47 @@
+    indice_pegado_costa = np.where(x == min(x))[0][0]
+    closed_x = _close_coordinate(x, indice_pegado_costa)
+    closed_y = _close_coordinate(y, indice_pegado_costa)
+    x_prueba = np.concatenate(
```
Por favor refactoriza las líneas 8-9 y 15-24 para que la 15 quedara algo así:
```suggestion
    x_prueba = _testing(limits[0:2], closed_x)
```
Creo que el nombre no es el correcto, pero no supe el objetivo de la función.

nepito commented on distdens/fillout.py:
```diff
@@ -0,0 +1,47 @@
+            np.flip(closed_x.reshape(len(closed_x), 1)),
+        ]
+    )
+    y_prueba = np.concatenate(
```
Por favor refactoriza las líneas 10-11 y 25-34 para que la 25 quedara algo así:
```suggestion
    y_prueba = _testing(limits[2:4], closed_y)
```

nepito reviewed (changes requested):
Listo @evaristor te dejé una prueba fallando para sugerirte una refactorizada

devarops commented:
Gracias por tu revisión, @nepito

devarops commented:
> Listo @evaristor te dejé una prueba fallando para sugerirte una refactorizada

@nepito No encontré la prueba fallando que me dejaste. ¿La pushaste o me la dejaste en tu máquina?

devarops committed:
Incorpora sugerencias de @nepito

Co-authored-by: nepito <braulio.rojas@islas.org.mx>

Commit:
Merge remote-tracking branch 'origin/feature/agrega_fillout' into feature/agrega_fillout

nepito commented:
> > Listo @evaristor te dejé una prueba fallando para sugerirte una refactorizada
> 
> @nepito No encontré la prueba fallando que me dejaste. ¿La pushaste o me la dejaste en tu máquina?

Aquí la había dejado. La acabo de _pushar_ ojalá te sirva @evaristor

devarops committed:
Importa función privada para refactorizar

Commit:
Regresa a la versión anterior antes de la función `_testing`

Commit:
Agrega la función `_flip_closed`

Commit:
Refactoriza las funciones

Commit:
Da formato con black

Commit:
borra prueba innesesaria

nepito reviewed (approved):
La aprobamos en un tomate @islasfernando  y @nepito
